packages/components/date-input/src/date-input-segment.tsx

```diff
@@ -22,14 +22,8 @@ export const DateInputSegment: React.FC<DateInputSegmentProps> = ({
   ...otherProps
 }) => {
   const ref = useRef(null);
-
   let {segmentProps} = useDateSegment(segment, state, ref);
 
-  // @ts-expect-error autoCapitalize is not a valid prop
-  // Removing autoCapitalize as it causes bugs in Firefox.
-  // See: https://github.com/adobe/react-spectrum/issues/5599
-  delete segmentProps.autoCapitalize;
```

Comment:
Since the `autoCapitalize` prop has been removed, I also removed this process.
https://github.com/adobe/react-spectrum/pull/6657